Some `burn-dataset` tests fail on `Windows`

**Describe the bug**

Running all `burn-dataset` tests through the `Windows` CI, we get two errors for some of them:

```
failures:

---- dataset::sqlite::tests::sqlite_writer_write stdout ----
thread 'dataset::sqlite::tests::sqlite_writer_write' panicked at 'Failed to set completed: PersistDbFile(Error { error: Os { code: 32, kind: Uncategorized, message: "The process cannot access the file because it is being used by another process." }, handle: Handle { id: 3, _marker: PhantomData<gix_tempfile::handle::Writable> } })', burn-dataset\src\dataset\sqlite.rs:791:32
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- dataset::sqlite::tests::sqlite_writer_write_multi_thread stdout ----
thread 'dataset::sqlite::tests::sqlite_writer_write_multi_thread' panicked at 'Should set completed successfully: PersistDbFile(Error { error: Os { code: 32, kind: Uncategorized, message: "The process cannot access the file because it is being used by another process." }, handle: Handle { id: 2, _marker: PhantomData<gix_tempfile::handle::Writable> } })', burn-dataset\src\dataset\sqlite.rs:839:14


failures:
    dataset::sqlite::tests::sqlite_writer_write
    dataset::sqlite::tests::sqlite_writer_write_multi_thread

test result: FAILED. 16 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.14s

error: test failed, to rerun pass `-p burn-dataset --lib`
error: process didn't exit successfully: `target\debug\xtask.exe run-checks std` (exit code: 101)
Error: Process completed with exit code 1.
```

For now, tests have been ignored on `Windows` in search of a possibile solution

--- a/burn-dataset/src/dataset/sqlite.rs
+++ b/burn-dataset/src/dataset/sqlite.rs
@@ -274,9 +274,7 @@ fn create_conn_pool<P: AsRef<Path>>(
         OpenFlags::SQLITE_OPEN_READ_ONLY
     };
 
-    // Create a connection pool and make sure the connections are read only
     let manager = SqliteConnectionManager::file(db_file).with_flags(sqlite_flags);
-
     Pool::new(manager).map_err(SqliteDatasetError::ConnectionPool)
 }
 

--- a/burn-dataset/src/dataset/sqlite.rs
+++ b/burn-dataset/src/dataset/sqlite.rs
@@ -560,6 +560,13 @@ where
     pub fn set_completed(&mut self) -> Result<()> {
         let mut is_completed = self.is_completed.write().unwrap();
 
+        // Force close the connection pool
+        // This is required on Windows platform where the connection pool prevents
+        // from persisting the db by renaming the temp file.
+        if let Some(pool) = self.conn_pool.take() {
+            std::mem::drop(pool);
+        }
+
         // Rename the database file from tmp to db
         let _file_result = self
             .db_file_tmp

